lib/index.js

```diff
@@ -12,23 +12,54 @@
       preflightContinue: false
     };
 
+  function varyHeadersOn(vary, headers) {
```

Comment:
@alex94puchades I'm a bit confused on what this function is doing. can you walk me through it?